File: android/src/main/kotlin/id/doran/jete_tws_sdk/DeviceInfoModel.kt
package id.doran.jete_tws_sdk

import com.bluetrum.devicemanager.models.DevicePower
import com.bluetrum.devicemanager.models.RemoteEqSetting


class DeviceInfoModel(
    val devicePower: DevicePower?,
    val deviceFirmwareVersion: Int?,
    val deviceName: String?,
    val deviceEqSetting: RemoteEqSetting?,
    val deviceKeySettings:Map<Int, Int>?,
    val deviceVolume:Byte?,
    val DevicePlayState:Boolean?,
    val DeviceWorkMode:Byte?,
    val deviceInEarStatus:Boolean?,
    val deviceLanguageSetting:Byte?,
    val deviceAutoAnswer:Boolean?,
    val deviceAncMode:Byte?,
    val deviceIsTws:Boolean?,
    val deviceTwsConnected:Boolean?,
    val deviceLedSwitch:Boolean?,
    val deviceFwChecksum:ByteArray?,
    val deviceAncGain:Int?,
    val deviceTransparencyGain:Int?,
    val deviceAncGainNum:Int?,
    val deviceTransparencyGainNum:Int?,
    val deviceRemoteEqSettings:List<RemoteEqSetting>?,
    val deviceLeftIsMainSide:Boolean?,
    val deviceProductColor:Int?,
    val deviceSoundEffect3d:Boolean?,
    val deviceCapacities:Int?,
    val deviceMaxPacketSize:Short?
)

File: ios/Classes/DeviceInfoModel.swift
//
//  DeviceInfoModel.swift
//  jete_tws_sdk
//


import Foundation

struct DeviceInfoModel {
    let devicePower: [String: Any]?
    let deviceFirmwareVersion: UInt32?
    let deviceName: String?
    let deviceEqSetting: [String: Any]?
    let deviceKeySettings: [[String: Any]]?
    let deviceVolume: UInt8?
    let devicePlayState: Bool?
    let deviceWorkMode: UInt8?
    let deviceInEarStatus: Bool?
    let deviceLanguageSetting: UInt8?
    let deviceAutoAnswer: Bool?
    let deviceAncMode: UInt8?
    let deviceIsTws: Bool?
    let deviceTwsConnected: Bool?
    let deviceLedSwitch: Bool?
    let deviceFwChecksum: Data?
    let deviceAncGain: Int?
    let deviceTransparencyGain: Int?
    let deviceAncGainNum: Int?
    let deviceTransparencyGainNum: Int?
    let deviceRemoteEqSettings: [[String: Any]]?
    let deviceLeftIsMainSide: Bool?
    let deviceProductColor: UInt8?
    let deviceSoundEffect3d: Bool?
    let deviceCapacities: UInt16?
    let deviceMaxPacketSize: UInt16?
}


